core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -136,6 +136,8 @@ private[spark] class AppStatusStore(
     store.read(classOf[StageDataWrapper], Array(stageId, stageAttemptId)).locality
   }
 
+  private def isInMemoryStore: Boolean = store.isInstanceOf[InMemoryStore] || listener.isDefined
```

Comment:
This check isn't whether it's an in-memory store, but whether you're running a live UI (vs. in-memory store in the SHS). Is that your intent? (The method name should match what you intend to check.)

Reply:
I checked if it is live UI in the first commit. 
As per the comment in https://github.com/apache/spark/pull/23088/files#diff-3bd1667878f7bda9a56f95e93a80b475R233 and https://github.com/apache/spark/pull/25369#discussion_r311479709, I change it to checking if it is InMemoryStore. 
But later on, I don't like the idea that live UI is inconsistent with SHS. So I raise a question.

Reply:
Ok, I looked at this again ignoring the PR title and this makes sense. Sorry for flip-flopping here, but could you put this in a local variable in `taskSummary` with a comment explaining it? e.g.

```
// SPARK-26119: we only want to consider successful tasks when calculating the metrics summary,
// but currently this is very expensive when using a disk store. So we only trigger the slower code
// path when we know we have all data in memory. This check is an approximation of when the know
// that the data will be in memory.
val isInMemoryStore = store.isInstanceOf[InMemoryStore] || listener.isDefined
```

Reply:
Thanks, I have added comments